src/ApiClient.php

```diff
@@ -171,70 +156,49 @@ private function createContainer(
             'client',
         ])));
 
-        // All services are not shared, meaning that a new instance is
+        // Endpoint services are not shared, meaning that a new instance is
         // created every time the service is requested from the container.
         // We're doing this because such a service might be called more than
         // once during the lifetime of a request, so internals set in a previous
         // usage may leak into the later usages.
+        $container->add('rest', Rest::class)
+            ->addArguments([
+                $httpClient,
+                $requestFactory,
+                $streamFactory,
+            ]);
         $container->add('main', MainEndpoint::class)
+            ->addArgument('rest')
             ->addArgument(new LiteralArgument($this->getConfigValue('mainApiEndpoint')));
         $container->add('referenceData', ReferenceDataEndpoint::class)
+            ->addArgument('rest')
             ->addArgument(new LiteralArgument($this->getConfigValue('referenceDataApiEndpoint')));
         $container->add('validate', ValidateEndpoint::class)
+            ->addArgument('rest')
             ->addArgument(new LiteralArgument($this->getConfigValue('validateApiEndpoint')));
         $container->add('requests', RequestsEndpoint::class)
+            ->addArgument('rest')
             ->addArgument(new LiteralArgument($this->getConfigValue('requestsApiEndpoint')));
         $container->add('identifier', IdentifierEndpoint::class)
+            ->addArgument('rest')
             ->addArgument(new LiteralArgument($this->getConfigValue('identifierApiEndpoint')));
         $container->add('status', StatusEndpoint::class)
+            ->addArgument('rest')
             ->addArgument(new LiteralArgument($this->getConfigValue('statusApiEndpoint')));
         $container->add('file', FileEndpoint::class)
+            ->addArgument('rest')
             ->addArgument(new LiteralArgument($this->getConfigValue('fileApiEndpoint')));
         $container->add('auth', TokenEndpoint::class)
+            ->addArgument('rest')
             ->addArguments([
                 new LiteralArgument($this->getConfigValue('tokenApiEndpoint')),
                 'token_config',
             ]);
 
-        // Inject the services into endpoints.
```

Comment:
This is not needed anymore, so perhaps we can close https://citnet.tech.ec.europa.eu/CITnet/jira/browse/D8AGE-363